R/calculate_stay.R

```diff
@@ -0,0 +1,32 @@
+#' Calculate total length of stay
+#'
+#' @description Calculate the total length of stay between start_date and end_date.
+#' If the end_date is missing then use the dummy discharge date.
+#'
+#' @param year The financial year in '1920' format
+#' @param start_date The admission/start date variable. e.g. record_keydate1
+#' @param end_date The discharge/end date variable. e.g record_keydate2
+#' @param sc_qtr The latest submitted quarter. e.g. sc_latest_submission
+#'
+#' @return a [tibble][tibble::tibble-package] with additional variable `stay`.
+#' If there is no end date use dummy discharge to calculate the total length of stay.
+#' @export
+#'
+#' @family date functions
+calculate_stay <- function(year, start_date, end_date, sc_qtr = NULL) {
+  lubridate::time_length(lubridate::interval(start_date, end_date), unit = "days")
+
+  if (is.na(end_date) & is.null(sc_qtr)) {
+    dummy_discharge <- dplyr::if_else(
+      is.na(end_date),
+      end_fy(year) + days(1),
+      end_date
+    )
+
+    lubridate::time_length(lubridate::interval(start_date, dummy_discharge), unit = "days")
+  } else if (is.na(end_date) & !is.null(sc_qtr)) {
+    qtr_end <- yq(sc_qtr) %m+% period(6, "months")
```

Comment:
```suggestion
    qtr_end <- yq(sc_qtr) %m+% period(3, "months")
```

Reply:
`yq` is working it out as the cal year. So 2017 Q1 is Jan-Mar 2017, whereas we want the FY year, so need to add 3 months (1 quarter) to it to make it correct. We then need to add another 3 months to get the end of the quarter.

Possibly worth making some functions for SC period dates? - Take SC `period` and return date e.g. `start_sc_quarter(period)` `end_sc_quarter(period)`. So in this case we'd use `end_sc_quarter(period) + lubridate::days(1)` (or equivalent)